### DISCUSSION TITLE
tests folder flake8

### DISCUSSION THREAD

**Discussion by arif-ali** [sosreport/sos#3513] [Q&A] [answered]
Going through some old PRs and such, we are having to pick up redundant code or variables not being using if people are updating or writing tests.

Is there any reason why we don't run flake8 against the `tests` folder? and fix any of the items there as part of our CI?

**Comment by TurboTurtle** [comment, marked as answer]
Mostly historical reasons - we never did before, so I didn't include them during the switch to avocado. That plus the idea at the time that tests don't get packaged. I'm now aware that at least _some_ Debian/Ubuntu packaging does in fact do that, so I have no objections to changing this and including it in the linting.

**Reply by arif-ali** [reply]
Thanks, makes sense.

I've started a PR, will do further on this too. My idea was that if we want to start writing more tests (which is Canonical folks intend to do) it would be good to have good code practice on the tests as well as the main code.

This will ensure we continue to have good practice across the whole codebase
